rfcs/011-dedicated-renku-git-user/011-dedicated-renku-git-user.md

```diff
@@ -0,0 +1,113 @@
+that is done by the KG. All this activity will now be associated with the Renkubot user rather
+than individual real users.
+
+## Design Detail
+
+### Helm Chart
+
+In the Helm chart we would add a hook that would create a user in the Gitlab deployment
+that Renku is connected to. The username and password will be saved in a secret. The KG
+would then have access to the secret and use it to authenticate with Gitlab.
+
+Alternatively the Helm chart can also accept existing user credentials. 
+
+### Gateway
+
+Either as part of this RFC or as an improvement the management of this secret / tokens for 
+the Renkubot user should be done by the Gateway and the KG will simply either ask the Gateway 
+for credentials or a Gateway's forward proxy will inject the right credentials in the right calls.
+
+This will also be helpful when we support multiple Gitlabs/Github. In this case the KG does
+not need to keep track which credentials to use for which Gitlab. The gateway can figure this
+out and inject the right credentials for the Renkubot user.
+
+### UI
+
+There are some changes needed in the UI for this. Namely:
+- To get a list of all projects for a user the UI would have to call Gitlab on behalf
+of the user. This is because the KG will now only know about projects to which the Renkubot
+user has been added. In other words, the KG will not know about any projects for which 
+KG integration is not active. It could figure this out for public Gitlab projects
+but it would be impossible to find private projects that are not in the KG.
+- Creating a project on Renku will have to include adding this Renkubot user to the project
+with `Reporter` permissions.
+- KG integration for a specific project can still be verified with the KG. In some cases
+the KG will simply not know anything at all for a project. In this case KG integration
+for this project is not enabled. Checking that the Renkubot user is in the project with the
+right permissions is not enough. It could be that the project has the Renkubot user but
+the KG has not fully finished indexing the project.
+- Enabling KG integration on projects that do not have it will have to be modified to also
+add the user to the project similarly to creating a project.
+- Disabling / removing KG integration results in removal of the project from the KG and removing
+the Renkubot user from the project.
+
+### KG
+
+Depending on when the Gateway refactoring is completed, the KG may not have to deal with the Renkubot
+credentials at all. This is the preferred option. But it may turn out that an initial implementation
+of this gives the Renkubot credentials directly to the KG at first.
+
+## Drawbacks
+
+1. The Renkubot user will have read access to many repos. If the credentials are stolen or leaked
+then they could be used to read the contents of private repos. 
+
+2. When this is rolled out we would automatically add the Renkubot user to all projects. This may
+be confusing to users.
```

Comment:
Well, all the repos already have a renkubot user :) However, each repo has a different one.

Reply:
All project maintainers will probably be notified that the bot user has been added. We will need to communicate this clearly to the users so that no one goes and removes it in response.